sw/device/silicon_creator/manuf/skus/earlgrey_a0/sival_bringup/ft_personalize.c

```diff
@@ -274,25 +362,126 @@ static status_t personalize_dice_certificates(ujson_t *uj) {
-      /*offset=*/0, endorsed_certs.uds_certificate_size / sizeof(uint32_t),
+      &kFlashCtrlInfoPageUdsCertificate, /*offset=*/0,
+      size_to_words(get_cert_size(endorsed_certs.uds_certificate)),
       endorsed_certs.uds_certificate));
   LOG_INFO("Imported DICE UDS certificate.");
 
+  // Now TPM certificates, get saved in the same flash INFO page.
+  uint32_t page_offset = 0;
+  size_t i;
+  unsigned char *tpm_certs[] = {
+      endorsed_certs.tpm_ek_certificate,
+      endorsed_certs.tpm_cek_certificate,
+      endorsed_certs.tpm_cik_certificate,
+  };
+
+  for (i = 0; i < ARRAYSIZE(tpm_certs); i++) {
+    const char *names[] = {"EK", "CEK", "CIK"};
+    // Number of words necessary for certificate storage.
+    uint32_t cert_size_words = size_to_words(get_cert_size(tpm_certs[i]));
+    uint32_t cert_size_bytes = cert_size_words * sizeof(uint32_t);
+
+    if ((page_offset + cert_size_bytes) > FLASH_CTRL_PARAM_BYTES_PER_PAGE) {
+      LOG_ERROR("TPM %s Certificate did not fit into the info page.", names[i]);
+      return OUT_OF_RANGE();
+    }
+    TRY(flash_ctrl_info_write(&kFlashCtrlInfoPageTpmCerts, page_offset,
+                              cert_size_words, tpm_certs[i]));
+    LOG_INFO("Imported TPM %s certificate.", names[i]);
+    page_offset += cert_size_bytes;
+
+    // Each certificate must be 8 bytes aligned.
+    page_offset = round_up_to(page_offset, 3);
+  }
+
+  uint32_t cert_size;
+  sha256_state_t state;
+  serdes_sha256_hash_t hash;
+
+  sha256_init(&state);
+  // Dice certificates, each in its own INFO page.
+  TRY(hash_certificate(&kFlashCtrlInfoPageUdsCertificate, 0, &state, NULL));
+  TRY(hash_certificate(&kFlashCtrlInfoPageCdi0Certificate, 0, &state, NULL));
+  TRY(hash_certificate(&kFlashCtrlInfoPageCdi1Certificate, 0, &state, NULL));
+
+  // TPM certificates, all in one INFO page.
+  page_offset = 0;
+  for (i = 0; i < ARRAYSIZE(tpm_certs);
+       i++) {  // There are three TPM certificates on the page.
+    TRY(hash_certificate(&kFlashCtrlInfoPageTpmCerts, page_offset, &state,
+                         &cert_size));
+    page_offset += size_to_words(cert_size) * sizeof(uint32_t);
+    page_offset = round_up_to(page_offset, 3);
+  }
+
+  TRY(sha256_final(&state, (uint32_t *)&hash));
+
+  // DO NOT CHANGE THE BELOW STRING without modifying the host code in
+  // sw/host/provisioning/ft_lib/src/lib.rs
+  LOG_INFO("Final hash: ");
+  RESP_OK(ujson_serialize_serdes_sha256_hash_t, uj, &hash);
+
+  // Also print the final hash on the console, first convert it into a hex
+  // string, two bytes per digit plus trailing zero.
+  char hash_string[sizeof(hash) * 2 + 1];
+  memset(hash_string, 0, sizeof(hash_string));
+  size_t room = sizeof(hash_string);
+
+  for (i = 0; i < sizeof(hash); i++) {
+    room -= base_snprintf(hash_string + sizeof(hash_string) - room, room,
+                          "%02x", ((unsigned char *)&hash)[i]);
+  }
+  LOG_INFO("Certs hash: %s", hash_string);
```

Comment:
```suggestion
  LOG_INFO("Sha256Sum of Certificates: 0x%08x%08x%08x%08x%08x%08x%08x%08x", hash[7],hash[6],hash[5],hash[4],hash[3],hash[2],hash[1],hash[0]);
```?

Reply:
amazing, it even prints it the same as the host :)
done